sycl/include/CL/sycl/platform.hpp

```diff
@@ -126,11 +126,12 @@ class __SYCL_EXPORT platform {
   /// given feature.
   bool has(aspect Aspect) const;
 
+  platform(shared_ptr_class<detail::platform_impl> impl) : impl(impl) {}
+
```

Comment:
Same here, why make this public?

Reply:
We have decided to undo this change and declare the function "make_platform()" as a friend of the platform class so that it can access this private constructor.

Reply:
small nit here .. the location of the private `platform` constructor got moved, so it shows up as an extraneous change in your PR.  Unless there's a specific reason to move the line, you should move it back to where it was before.